app/src/main/java/org/xbmc/kore/ui/sections/remote/PlaylistFragment.java

```diff
@@ -83,21 +89,30 @@
     private Handler callbackHandler = new Handler();
 
     /**
-     * The current active player id
+     * Playlist adapter
      */
-    private int currentActivePlayerId = -1;
+    private PlayListAdapter playListAdapter;
 
-    /**
-     * Current playlist
-     */
-    private int currentPlaylistId = -1;
+    private Unbinder unbinder;
 
     /**
-     * Playlist adapter
+     * Last call results
      */
-    private PlayListAdapter playListAdapter;
+    private ListType.ItemsAll lastGetItemResult = null;
+    private PlayerType.GetActivePlayersReturnType lastGetActivePlayerResult;
+    private HashMap<String, PlaylistHolder> playlists = new HashMap<>();
+
+    private enum PLAYER_STATE {
+        CONNECTION_ERROR,
+        NO_RESULTS_YET,
+        PLAYING,
+        PAUSED,
+        STOPPED
+    }
```

Comment:
Similar to what's on `HostConnectionObserver.PlayerEventsObserver`?

Reply:
Indeed. So if I understand correctly I can refactor this and make the `HostConnectionObserver` actually maintain the player state. If `PlaylistFragment` needs the player state it should ask the `HostConnectionObserver`. Correct?

Reply:
I wasn't going that far, i was just pointing out that you could use the same enums.
I'm not sure where you're going, but my preference is to not overload `HostConnectionObserver` with much more to do, and if that means that some other classes need to maintain state and not rely on it to do everything, so be it.